-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-32761/20914

ERROR: Build errors found during clang-tidy run.
```
DQMServices/Core/src/MonitorElement.cc:907:24: error: out-of-line definition of 'setStatOverflows' does not match any declaration in 'dqm::impl::MonitorElement' [clang-diagnostic-error]
  void MonitorElement::setStatOverflows(bool value) {
                       ^
Suppressed 1377 warnings (1376 in non-user code, 1 with check filters).
--
gmake: *** [config/SCRAM/GMake/Makefile.coderules:128: code-checks] Error 2
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2
```

--- a/DQM/SiPixelPhase1Common/src/HistogramManager.cc
+++ b/DQM/SiPixelPhase1Common/src/HistogramManager.cc
@@ -296,7 +296,8 @@ void HistogramManager::book(DQMStore::IBooker& iBooker, edm::EventSetup const& i
     GeometryInterface::Value binwidth_y = 0;
     std::string title, xlabel, ylabel, zlabel;
     bool do_profile = false;
-    bool statsOverflows = true;;
+    bool statsOverflows = true;
+    ;
   };
   std::map<GeometryInterface::Values, MEInfo> toBeBooked;
 
@@ -344,7 +345,7 @@ void HistogramManager::book(DQMStore::IBooker& iBooker, edm::EventSetup const& i
         // create new histo
         MEInfo& mei = toBeBooked[significantvalues];
         mei.title = this->title;
-	mei.statsOverflows = this->statsOverflows;
+        mei.statsOverflows = this->statsOverflows;
         if (bookCounters)
           mei.title =
               "Number of " + mei.title + " per Event and " + geometryInterface.pretty(*(s.steps[0].columns.end() - 1));
@@ -605,7 +606,6 @@ void HistogramManager::executeGroupBy(SummationStep const& step,
       new_histo.th1->Add(th1);
     }
     new_histo.me->setStatOverflows(e.second.me->getStatOverflows());
-
   }
   t.swap(out);
 }
@@ -687,7 +687,7 @@ void HistogramManager::executeExtend(SummationStep const& step,
       } else {
         assert(!"Reduction type not supported");
       }
-      new_histo.me->setStatOverflows(e.second.me->getStatOverflows());	  
+      new_histo.me->setStatOverflows(e.second.me->getStatOverflows());
     } else {
       assert(!"2D extend not implemented in harvesting.");
     }

--- a/DQMServices/Core/interface/MonitorElement.h
+++ b/DQMServices/Core/interface/MonitorElement.h
@@ -427,9 +427,9 @@ namespace dqm::impl {
     virtual void setCanExtend(unsigned int value);
     // We should decide if we support this (or make it default)
     DQM_DEPRECATED
-    virtual void setStatOverflows(unsigned int value);
+    virtual void setStatOverflows(bool value);
     virtual bool getStatOverflows();
-   
+
     // these should be non-const, since they are potentially not thread-safe
     virtual TObject const *getRootObject() const;
     virtual TH1 *getTH1();

--- a/DQMServices/Core/src/MonitorElement.cc
+++ b/DQMServices/Core/src/MonitorElement.cc
@@ -912,9 +912,9 @@ namespace dqm::impl {
       access.value.object_->SetStatOverflows(TH1::kIgnore);
   }
 
-  bool MonitorElement::getStatOverflows(){
+  bool MonitorElement::getStatOverflows() {
     auto access = this->accessMut();
-    auto value=access.value.object_->GetStatOverflows();
+    auto value = access.value.object_->GetStatOverflows();
     if (value == TH1::kConsider)
       return true;
     else

